### PR TITLE
Verify that each named file is unique.

### PR DESCRIPTION
**Pull request by DrMarcII**

### CODE REVIEW

File: go/metadata/merger_test.sh [hunk 1]
```diff
@@ -16,11 +16,43 @@
...
+
 $TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/merger --output $TEST_TMPDIR/out.json \
 	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/chrome-linux.json \
 	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/android-browser-gingerbread-nexus-s.json
 
 diff $TEST_TMPDIR/out.json \
-	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/merger-result.json
\ No newline at end of file
+	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/merger-result.json
+if [ $? != 0 ]; then
```
Comment (juangj) [inline go/metadata/merger_test.sh]: Prefer [[ to [.
https://google.github.io/styleguide/shell.xml#Test,_%5B_and_%5B%5B

File: go/metadata/merger_test.sh [hunk 2]
```diff
@@ -16,11 +16,43 @@
...
+	echo "Merge result didn't match."
+	error=1
+fi
+
+$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/merger --output $TEST_TMPDIR/out2.json \
+	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/named-files1.json \
+	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/named-files1.json
+if [ $? != 0 ]; then
+	echo "Expected successful named files merge failed."
```
Comment (juangj) [inline go/metadata/merger_test.sh]: I find this error message hard to parse. This is the intent, right?
> [I] expected [a] successful named-files-merge [but I tried it and it] failed

How about more explicit statements of what you expect?
* Merging named files maps with conflicting names should fail.
* bad-named-files.json should fail to load.

File: go/metadata/metadata_test.go [hunk 3]
```diff
@@ -209,3 +227,270 @@ func TestEquals(t *testing.T) {
...
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := validateNoDuplicateNamedFiles(tc.namedFiles, tc.archives)
+			if err == nil && tc.err {
+				t.Error("Got nil, expected err")
+			}
+			if err != nil && !tc.err {
+				t.Errorf("Exoected nil, got %v", err)
```
Comment (juangj) [inline go/metadata/metadata_test.go]: s/Exoected/Expected/

File: go/metadata/merger_test.sh (hunk 1 above)
Comment (DrMarcII) [inline go/metadata/merger_test.sh]: Done.

File: go/metadata/merger_test.sh (hunk 2 above)
Comment (DrMarcII) [inline go/metadata/merger_test.sh]: Done.

File: go/metadata/metadata_test.go (hunk 3 above)
Comment (DrMarcII) [inline go/metadata/metadata_test.go]: Done.
